### PR TITLE
use ES6 syntax

### PR DESCRIPTION
**Pull request by davidchambers**
sanctuary-js/sanctuary#687

### CODE REVIEW

File: bin/sanctuary-benchmark [hunk 1]
```diff
@@ -2,74 +2,73 @@
...
-var path = require ('path');
+const path = require ('path');
 
 
-var dir = path.resolve (process.cwd (), 'bench');
+const dir = path.resolve (process.cwd (), 'bench');
 
-var help = 'Create a bench directory in this project to get started\n';
+let help = `Create a bench directory in this project to get started
+`;
```
Comment (davidchambers) [inline bin/sanctuary-benchmark]: I'm equally happy to stick with `\n` in this case. Do you have a preference, Aldwin?

File: bin/sanctuary-benchmark [hunk 2]
```diff
@@ -2,74 +2,73 @@
...
+
+Example
+
+  module.exports = require ('sanctuary-benchmark') (
+    require ('ramda'),
+    require ('sanctuary'),
+    {left: 'Ramda', right: 'Sanctuary'},
+    {map: lib => lib.map (x => x + 1) ([1, 2, 3])}
+  );
+`;
```
Comment (davidchambers) [inline bin/sanctuary-benchmark:40]: :tada:

File: index.js [hunk 3]
```diff
@@ -161,78 +155,74 @@ function repeat(n, s) {
...
-        var difference = Math.abs ((newRes.hz - oldRes.hz) /
-                                   ((oldRes.hz + newRes.hz) / 2) /
-                                   2);
-        var isPositive = difference > significantDifference && change > 0;
-        var isNegative = difference > significantDifference && change < 0;
-
-        var highlight = colors && isPositive ? green :
-                        colors && isNegative ? red :
-                        identity;
+        const {0: oldRes, 1: newRes} = this;
```
Comment (davidchambers) [inline index.js:215]: I spent significant time debugging an error that resulted from trying `[oldRes, newRes]` here.

File: test/index.js [hunk 4]
```diff
@@ -15,76 +15,78 @@ function eq(actual, expected) {
...
-          JSON.stringify (line) + '\n\n' +
-          'It does not match expression:\n\n' +
-          expectations[idx].toString ()
+          `A printed line did not meet expectations:
+
+${JSON.stringify (line)}
+
+It does not match expression:
+
+${expectations[idx]}`
```
Comment (davidchambers) [inline test/index.js]: This indentation bothers me. How do you feel about this, Aldwin?

File: bin/sanctuary-benchmark (hunk 1 above)
Comment (Avaq) [reply bin/sanctuary-benchmark]: I prefer the `\n`. The new version looks like a mistake.

File: test/index.js (hunk 4 above)
Comment (Avaq) [reply test/index.js]: It bothers me too. You could use the [`deindent`](https://www.npmjs.com/package/deindent) package, or sometimes I also just fall back on something like:

```js
    `A printed line did not meet expectations:\n\n${
      JSON.stringify (line)
    }\n\nIt does not match expression:\n\n${
      expectations[idx]
    }`
```

**Review by Avaq — APPROVED** [review]

**Pull request merged**